pkg/reconciler/configuration/configuration.go

```diff
@@ -215,33 +215,36 @@ func (c *Reconciler) findAndSetLatestReadyRevision(config *v1alpha1.Configuratio
 
 // getSortedCreatedRevisions returns the list of created revisions sorted in descending
 // generation order between the generation of the latest ready revision and config's generation (both inclusive).
-func (c *Reconciler) getSortedCreatedRevisions(config *v1alpha1.Configuration) ([]*v1alpha1.Revision, error) {
+func (c *Reconciler) getSortedCreatedRevisions(ctx context.Context, config *v1alpha1.Configuration) ([]*v1alpha1.Revision, error) {
+	logger := logging.FromContext(ctx)
 	lister := c.revisionLister.Revisions(config.Namespace)
 	configSelector := labels.SelectorFromSet(map[string]string{
 		serving.ConfigurationLabelKey: config.Name,
 	})
 	if config.Status.LatestReadyRevisionName != "" {
 		lrr, err := lister.Get(config.Status.LatestReadyRevisionName)
+		// Record the error and continue because we still want to set the LRR to the correct revision
 		if err != nil {
-			return nil, err
-		}
-		start := lrr.Generation
-		var generations []string
-		for i := start; i <= int64(config.Generation); i++ {
-			generations = append(generations, strconv.FormatInt(i, 10))
-		}
+			logger.Errorf("Error getting latest ready revision %q: %v", config.Status.LatestReadyRevisionName, err)
+		} else {
+			start := lrr.Generation
+			var generations []string
+			for i := start; i <= int64(config.Generation); i++ {
+				generations = append(generations, strconv.FormatInt(i, 10))
+			}
 
-		// Add an "In" filter so that the configurations we get back from List have generation
-		// in range (config's latest ready generation, config's generation]
-		generationKey := serving.ConfigurationGenerationLabelKey
-		inReq, err := labels.NewRequirement(generationKey,
-			selection.In,
-			generations,
-		)
-		if err != nil {
-			return nil, err
+			// Add an "In" filter so that the configurations we get back from List have generation
+			// in range (config's latest ready generation, config's generation]
+			generationKey := serving.ConfigurationGenerationLabelKey
+			inReq, err := labels.NewRequirement(generationKey,
+				selection.In,
+				generations,
+			)
+			if err != nil {
+				return nil, err
```

Comment:
You also have an opportunity to exit out of this function here without returning a list of eligible Revisions.

IIUC this whole section is just limiting the search space down to Revisions between the previous Ready and the latest Created. Why not fallback to the broader search in this case as well?